client/src/reducers/teaReducers.js

```diff
@@ -1,20 +1,37 @@
-import { Set, Map } from "immutable";
-
-export default (state = [], action) => {
+export default (state = { allTeas: {}, teaIDs: [] }, action) => {
   switch (action.type) {
     case "ADD_TEA":
-      return [...state, action.payload];
-    case "DELETE_TEA":
-      return state.filter(t => t.id !== action.payload.id);
+      let addAllTeas = {
+        ...state.allTeas,
+        [action.payload.id]: action.payload
+      };
+      let addIDsArr = [...state.teaIDs, action.payload.id];
```

Comment:
I'd still ensure your IDs array is only returning unique values, to handle the case where an ADD_TEA action is inadvertently called with an existing tea.  That would be a bug, but this is a way to minimize the effects of that bug.  If you want to ensure that a bug like that doesn't go unnoticed, you could even raise an error here if a duplicate ID is detected, though in my opinion that is overkill.